src/Microsoft.SqlTools.ServiceLayer/QueryExecution/SpecialAction.cs

```diff
@@ -0,0 +1,125 @@
+// 
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution
+{
+    /// <summary>
+    /// Class that represents a Special Action which occured by user request during the query 
+    /// </summary>
+    public class SpecialAction {
```

Comment:
I'm having a really hard time wrapping my head around this class. If these are just boolean flags that we'd set as part of execution, why can't they just be part of the resultset object itself? If there's some kind of action that needs to be taken, I think it should be setup more like a SpecialAction list and have the SpecialAction be an interface that has a method for the action to take.